### PR TITLE
Refactor offenderService and increase test coverage

### PR DESCRIPTION
**Pull request by LucasCairns**

### CODE REVIEW

File: server/routes/index.js [hunk 1]
```diff
@@ -36,17 +43,32 @@ const createIndexRouter = ({ logger, hubFeaturedContentService, config }) => {
...
         personalInformation,
-        userName: userName ? `Hi, ${userName}` : null,
         establishmentId,
         returnUrl: req.originalUrl,
       };
 
+      const data = {};
+
+      if (req.user) {
+        const userName = req.user && `Hi, ${req.user.getFullName()}`;
```
Comment (psoleckimoj) [inline server/routes/index.js:53]: `Hi, ` isn't part of their username, it's a display thing so should be separate. Either in the template or a separate variable.

File: server/utils/string.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  return amount && currency
+    ? new Intl.NumberFormat('en-GB', {
+        style: 'currency',
+        currency,
+      }).format(amount)
+    : placeHolder;
+};
+
+module.exports = {
+  fullNameOr,
```
Comment (psoleckimoj) [inline server/utils/string.js:18]: Not gonna not LGTM this but I've never been a fan of function names ending half way through a sentence as they aren't self descriptive, as in Or what? They only make sense if you know what the args are. Same with For something as well, you're passing something in so it has to be For those args anyway.

File: server/routes/index.js (hunk 1 above)
Comment (psoleckimoj) [reply server/routes/index.js:53]: Actually, this would be better if we rename the var being passed in to `userNameDisplay` or something to indicate that?

Comment (LucasCairns) [reply server/routes/index.js]: Resolved, updated to `welcomeMessage`

File: server/utils/string.js (hunk 2 above)
Comment (LucasCairns) [reply server/utils/string.js]: Renamed to `..OrDefault` to be more explicit with intent

**Review by psoleckimoj — APPROVED** [review]
LGTM!

**Pull request merged**